src/platform_impl/macos/view.rs

```diff
@@ -910,13 +910,17 @@ fn mouse_motion(this: &Object, event: id) {
         let view_point = view.convertPoint_fromView_(window_point, nil);
         let view_rect = NSView::frame(view);
 
+        let mouse_buttons_down: NSInteger = msg_send![class!(NSEvent), pressedMouseButtons];
```

Comment:
Why put this outside of the conditional? Seems like a waste to do this when you don't need to?

Reply:
You can of course keep this outside and just add `|| mouse_button_down == 0`, which would reduce the level of nesting. Perhaps that was the original intention of putting it here?

Reply:
Moved the variable inside the if body.

It's possible to merge both conditions, but the result is (subjectively) harder to read because of all the brackets.

Reply:
Since there's also no clean way to inline the msg_send! directly, meaning that this variation also has a slight performance advantage in the general case of sending events, I agree that it's the best choice.